firebase-vscode/webviews/SidebarApp.tsx

```diff
@@ -0,0 +1,164 @@
+import { VSCodeButton } from "@vscode/webview-ui-toolkit/react";
+import React, { useEffect, useState } from "react";
+import { Spacer } from "./components/ui/Spacer";
+import { Body } from "./components/ui/Text";
+import { broker } from "./globals/html-broker";
+import { User } from "../../src/types/auth";
+import { FirebaseRC } from "../../src/firebaserc";
+import { PanelSection } from "./components/ui/PanelSection";
+import { AccountSection } from "./components/AccountSection";
+import { ProjectSection } from "./components/ProjectSection";
+import { FirebaseConfig } from "../../src/firebaseConfig";
+import { ServiceAccountUser } from "../common/types";
+import { DeployPanel } from "./components/DeployPanel";
+import { HostingState } from "./webview-types";
+import { ChannelWithId } from "./messaging/types";
+
+export function SidebarApp() {
+  const [projectId, setProjectId] = useState<string | null>(null);
+  const [hostingState, setHostingState] = useState<HostingState>(null);
+  const [env, setEnv] = useState<{ isMonospace: boolean }>();
+  const [channels, setChannels] = useState<ChannelWithId[]>(null);
+  const [userEmail, setUserEmail] = useState<string | null>(null);
+  /**
+   * null - has not finished checking yet
+   * empty array - finished checking, no users logged in
+   * non-empty array - contains logged in users
+   */
+  const [allUsers, setAllUsers] = useState<Array<
+    ServiceAccountUser | User
+  > | null>(null);
+  const [isHostingOnboarded, setHostingOnboarded] = useState<boolean>(false);
+
+  useEffect(() => {
+    console.log("loading SidebarApp component");
```

Comment:
A handful of console.log in this file

Reply:
I made a logger util for webviews as well that passes the messages to the extension process for debug logging.